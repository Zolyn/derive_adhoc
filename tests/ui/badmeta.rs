use derive_adhoc::{derive_adhoc, Adhoc};

#[derive(Adhoc)]
#[adhoc(wrong = "{")]
#[adhoc(dot = ".")]
struct DataType {}

derive_adhoc! {
    DataType:

    const K: () = ${tmeta(wrong)}
}
// Invoke derive_adhoc! a second time, because (empirically) the
// attempt to parse "{" causes some kind of nonlocal exit:
// syn::LitStr::parse is called but doesn't return.  And then, something
// arranges to report only the *last* such error.  But we want to see
// all the errors, so invoke the macro separately.

derive_adhoc! {
    DataType:

    struct ${paste Bad ${tmeta(dot) as str}};
}

fn main() {}
