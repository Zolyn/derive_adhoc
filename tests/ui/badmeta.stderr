error: this file contains an unclosed delimiter
  --> ui/badmeta.rs:3:10
   |
3  |   #[derive(Adhoc)]
   |            ^^^^^ unclosed delimiter
...
8  | / derive_adhoc! {
9  | |     DataType:
10 | |
11 | |     const K: () = ${tmeta(wrong)}
12 | | }
   | |_- in this macro invocation
   |
   = note: this error originates in the macro `::derive_adhoc::derive_adhoc_expand` which comes from the expansion of the macro `derive_adhoc` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected one of `.`, `;`, `?`, or an operator, found `<eof>`
 --> ui/badmeta.rs:4:17
  |
4 | #[adhoc(wrong = "{")]
  |                 ^^^ expected one of `.`, `;`, `?`, or an operator

error: pasted identifier "Bad." is invalid
  --> ui/badmeta.rs:22:14
   |
22 |     struct ${paste Bad ${tmeta(dot) as str}};
   |              ^^^^^

error: probably-invalid input to identifier pasting
  --> ui/badmeta.rs:5:15
   |
5  |   #[adhoc(dot = ".")]
   |                 ^^^
...
19 | / derive_adhoc! {
20 | |     DataType:
21 | |
22 | |     struct ${paste Bad ${tmeta(dot) as str}};
23 | | }
   | |_- in this macro invocation
   |
   = note: this error originates in the macro `derive_adhoc_driver_DataType` which comes from the expansion of the macro `derive_adhoc` (in Nightly builds, run with -Z macro-backtrace for more info)
